statsprinter.go

```diff
@@ -352,86 +431,62 @@ func (j *statsJsonPrinter) printStatistics() {
 		calcLongestDowntime(j.stats, j.lastUnsuccessfulProbe)
 	}
 
-	/* longest uptime stats */
-	j.printLongestUptime()
-
-	/* longest downtime stats */
-	j.printLongestDowntime()
-
-	/* resolve retry stats */
-	if !j.isIP {
-		j.printRetryResolveStats()
+	if j.longestUptime.duration != 0 {
+		data.LongestUptime = j.longestUptime.duration
+		data.LongestUptimeStart = &j.longestUptime.start
+		data.LongestUptimeEnd = &j.longestUptime.end
 	}
 
-	/* latency stats.*/
-	jsonPrintf("rtt min/avg/max: %.3f/%.3f/%.3f", rttResults.min, rttResults.average, rttResults.max)
-
-	/* duration stats */
-	j.printDurationStats()
-}
-
-/* Print TCP probe replies according to our policies in JSON format */
-func (j *statsJsonPrinter) printReply(replyMsg replyMsg) {
-	if j.isIP {
-		if replyMsg.msg == noReply {
-			jsonPrintf("%s from %s on port %d TCP_conn=%d",
-				replyMsg.msg, j.ip, j.port, j.totalUnsuccessfulProbes)
-		} else {
-			jsonPrintf("%s from %s on port %d TCP_conn=%d time=%.3f ms",
-				replyMsg.msg, j.ip, j.port, j.totalSuccessfulProbes, replyMsg.rtt)
-		}
-	} else {
-		if replyMsg.msg == noReply {
-			jsonPrintf("%s from %s (%s) on port %d TCP_conn=%d",
-				replyMsg.msg, j.hostname, j.ip, j.port, j.totalUnsuccessfulProbes)
-		} else {
-			jsonPrintf("%s from %s (%s) on port %d TCP_conn=%d time=%.3f ms",
-				replyMsg.msg, j.hostname, j.ip, j.port, j.totalSuccessfulProbes, replyMsg.rtt)
-		}
+	if j.longestDowntime.duration != 0 {
+		data.LongestDowntime = j.longestDowntime.duration
+		data.LongestDowntimeStart = &j.longestDowntime.start
+		data.LongestDowntimeEnd = &j.longestDowntime.end
 	}
-}
-
-/* Print the total downtime in JSON format */
-func (j *statsJsonPrinter) printTotalDownTime(now time.Time) {
-	latestDowntimeDuration := time.Since(j.startOfDowntime).Seconds()
-	calculatedDowntime := calcTime(uint(math.Ceil(latestDowntimeDuration)))
-
-	jsonPrintf("No response received for %s", calculatedDowntime)
-}
 
-/* Print the longest uptime in JSON format */
-func (j *statsJsonPrinter) printLongestUptime() {
-	if j.longestUptime.duration == 0 {
-		return
+	if !j.isIP {
+		data.HostnameResolveTries = j.retriedHostnameResolves
 	}
 
-	uptime := calcTime(uint(math.Ceil(j.longestUptime.duration)))
-	longestUptimeStart := j.longestUptime.start.Format(timeFormat)
-	longestUptimeEnd := j.longestUptime.end.Format(timeFormat)
-
-	jsonPrintf("longest consecutive uptime: %v from %v to %v", uptime, longestUptimeStart, longestUptimeEnd)
-}
-
-/* Print the longest downtime in JSON format */
-func (j *statsJsonPrinter) printLongestDowntime() {
-	if j.longestDowntime.duration == 0 {
-		return
+	latencyStats := findMinAvgMaxRttTime(j.rtt)
+	if latencyStats.hasResults {
+		data.LatencyMin = latencyStats.min
+		data.LatencyAvg = latencyStats.average
+		data.LatencyMax = latencyStats.max
```

Comment:
These and the longest up/downtime are printing more than 3 floating point fractions. Let's have them to be aligned with the normal output, three fraction points.

Reply:
Done. Updated TotalDuration as well 
<img width="735" alt="image" src="https://github.com/pouriyajamshidi/tcping/assets/5007271/9dcb3407-f28b-4f3e-82f3-e223a98c1cf0">

Reply:
Well done. Let's merge this in.